src/backend/distributed/utils/citus_clauses.c

```diff
@@ -452,3 +461,40 @@ CitusIsMutableFunction(Node *node)
 
 	return false;
 }
+
+
+/* FixFunctionArguments applies expand_function_arguments to all function calls. */
+static void
+FixFunctionArguments(Node *expr)
+{
+	FixFunctionArgumentsWalker(expr, NULL);
+}
+
+
+/* FixFunctionArgumentsWalker is the helper function for fix_funcargs. */
+static bool
+FixFunctionArgumentsWalker(Node *expr, void *context)
+{
+	if (expr == NULL)
+	{
+		return false;
+	}
+
+	if (IsA(expr, FuncExpr))
```

Comment:
Would it make sense to put this if check inside at the beginning of `PartiallyEvaluateExpression` so that we dont traverse the tree multiple times?

Reply:
The function call could be deeper than the initial node of `PartiallyEvaluateExpression`

For example, returning `funcExpr` in this `if` rather than continuing to recurse causes the `f(f())` test to segfault

Reply:
um, so in that case we would handle the first node but the following parts could be going deeper than that and in that case they might encounter nodes with default values not yet evaluated, which might cause the crash.